cpp/arcticdb/version/version_core.cpp

```diff
@@ -469,7 +469,21 @@ std::vector<SliceAndKey> read_and_process(
     return collect_segments(std::move(merged));
 }
 
-SegmentInMemory read_direct(const std::shared_ptr<Store>& store,
+SegmentInMemory get_future_frame_impl(SegmentInMemory frame){
+    util::print_total_mem_usage(__FILE__, __LINE__, __FUNCTION__);
+    return frame;
+}
+
+
+folly::Future<SegmentInMemory> get_future_frame(folly::Future<std::vector<VariantKey>>&& fut, SegmentInMemory& frame){
+    return std::move(fut).via(&async::cpu_executor()).thenValue(
+                        [&frame](std::vector<VariantKey>&&){
+                            return get_future_frame_impl(frame);
+                        }
+                    );
```

Comment:
![](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRc1Fs1QOXizwsc7opHi0DXGRK9L5AEE_X1K2F9BPFfMyGM_-mvU_bEhRx9xRSZx9hw1Fw&usqp=CAU)

Reply:
removed this function